WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -161,27 +162,14 @@ private extension ItemListView {
         }
     }
 
-    @ViewBuilder
-    func listRow(item: POSItem) -> some View {
-        switch item {
-        case .simpleProduct(let simpleProduct):
-            Button {
-                posModel.addToCart(simpleProduct)
-            } label: {
-                SimpleProductCardView(product: simpleProduct)
-            }
-        case .parentProduct(let parentProduct):
-            Button {
-                print("Tapped parent product")
-            } label: {
-                ParentProductCardView(parentProduct: parentProduct)
-            }
-        case .variation(let variation):
-            Button(action: {
-                print("Tapped variation \(variation.name)")
-            }, label: {
-                VariationCardView(variation: variation)
-            })
+    func childListView(parentItem: POSItem) -> some View {
+        guard case let .parentProduct(parentProduct) = parentItem else {
+            assertionFailure("Unexpected child view for parent item: \(parentItem)")
+            return AnyView(EmptyView())
+        }
+        switch parentProduct.type {
+        case .variable:
+            return AnyView(VariationItemList(parentItem: parentItem, parentProduct: parentProduct))
```

Comment:
By having a general `ItemList` which can take any `POSItem` as the parent, we can avoid using `AnyView` here, which would be a theoretical performance improvement. I don't think it will make much difference to performance, but it'd also be easier to read.

`ItemList` would need to handle getting an inappropriate parent, but that chould be shown as an error state, even if it shouldn't really happen.

Reply:
Yea good point, I remember reading about performance impact from using `AnyView`. If I use `@ViewBuilder` and don't return a view with the guard statement, `AnyView` can be avoided. Updated in https://github.com/woocommerce/woocommerce-ios/pull/14786/commits/76e09242323a1b0e2653aff12713791d701c344c.